### PR TITLE
[FIX] base: fix exception retro-compatibility-aid

### PR DESCRIPTION
**Pull request by Julien00859**
Restore the old signature of except_orm.

opw-2354525
opw-2355980

### CODE REVIEW

**Comment by Julien00859** [comment]
You really should remove all those deprecated crap from internal. It wouldn't take long, doesn't need to be done *now* but please at least fill a task in your team's backlog :/

**Comment by Julien00859** [comment]
Also I have no way to test it works

**Comment by tbe-odoo** [comment]
> You really should remove all those deprecated crap from internal. It wouldn't take long, doesn't need to be done _now_ but please at least fill a task in your team's backlog :/

You should know that we are running multiple versions of Odoo with the same code. From 8.0 to 14.0 in our SaaS. This might look like crap for you, but that's what backward compatibility is about.

**Comment by Julien00859** [comment]
Is there nothing we can do about `saas_trial` code ?

**Comment by tbe-odoo** [comment]
> Is there nothing we can do about `saas_trial` code ?

The only thing that can be done for `saas_trial` to work in multiple versions, is to keep a layer of backward compatibility, as it is done in multiple place on Odoo.


c.f.
https://github.com/odoo/odoo/blob/2444fde7f852787d87589a93c4bc385d76ff20e9/odoo/modules/module.py#L25
https://github.com/odoo/odoo/blob/4b38cc6590abcb58a0ca102a06063eb3db7ac1f4/odoo/addons/base/models/decimal_precision.py#L71
https://github.com/odoo/odoo/blob/2444fde7f852787d87589a93c4bc385d76ff20e9/odoo/modules/module.py#L55 (btw you brought back the backward compatibility in here)

**Comment by Julien00859** [comment]
If I understand it clearly, you are using a v12+ (=recent) ORM to run the SaaS right ? You use the same ORM to run all the databases, so v8 customers are using a recent ORM so we need to keep the backward compatibility stuff for them. I understand it clearly.

By ORM I mean all the bootstrapping code, all the stuff in community/odoo/.

What I don't understand is why saas_trial and other internal modules are still using some v8 deprecated APIs. Are those modules bind to the Odoo version the customer is using ? Like you must run a v8 saas_trial for a v8 database even if the ORM is more recent ? That there is *noway* to upgrade that hypothetical v8-only `saas_trial` so it doesn't use deprecated APIs anymore.

I fail to understand why custom modules like saas_trial would absolutely need to use a v8 API if it is the same (recent) ORM that is used all over. I agree changing `except_orm` v8 exceptions to the "new" semantic ones is not critical but citing my message above :

> It wouldn't take long, doesn't need to be done now but please at least fill a task in your team's backlog :/

Btw, I don't have r+ and they are customers waiting

**Comment by tbe-odoo** [comment]
>  customers are using a recent ORM so we n

Hello, the v8 databases are running in the 8.0 branches (`odoo`/`enterprise`) but `saas_trial` run on multiple versions.
Meaning, the `saas_trial` code should be able to run on all versions.

For example, in `saas_trial` module we cannot do

```python
from odoo import tools
```

For a single reason, in older versions `odoo` is undefined we can only import from `openerp`.
That's why even on newer versions we still support importing from `openerp` as a backward compatibility measure.

So on `saas_trial` doing the import this way works.
```python
from openerp import tools
```
To summarize, trial modules on SaaS are written in a way that it can work on all Odoo versions, to do so, they rely on backward compatibility features inside Odoo, otherwise it wouldn't be do-able.

**Comment by tbe-odoo** [comment]
> If I understand it clearly, you are using a v12+ (=recent) ORM to run the SaaS right ? You use the same ORM to run all the databases, so v8 customers are using a recent ORM so we need to keep the backward compatibility stuff for them. I understand it clearly.
> 
> By ORM I mean all the bootstrapping code, all the stuff in community/odoo/.
> 
> What I don't understand is why saas_trial and other internal modules are still using some v8 deprecated APIs. Are those modules bind to the Odoo version the customer is using ? Like you must run a v8 saas_trial for a v8 database even if the ORM is more recent ? That there is _noway_ to upgrade that hypothetical v8-only `saas_trial` so it doesn't use deprecated APIs anymore.
> 
> I fail to understand why custom modules like saas_trial would absolutely need to use a v8 API if it is the same (recent) ORM that is used all over. I agree changing `except_orm` v8 exceptions to the "new" semantic ones is not critical but citing my message above :
> 
> > It wouldn't take long, doesn't need to be done now but please at least fill a task in your team's backlog :/
> 
> Btw, I don't have r+ and they are customers waiting

I'll send you more details in private explaining how our SaaS work

**Comment by tbe-odoo** [comment]
@robodoo r+
